spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/outbound/MongoDbOutboundGatewayTests.java

```diff
@@ -0,0 +1,305 @@
+		this.mongoTemplate.save(this.createPerson("Gary"), COLLECTION_NAME);
+		this.mongoTemplate.save(this.createPerson("Oleg"), COLLECTION_NAME);
+		this.mongoTemplate.save(this.createPerson("Xavi"), COLLECTION_NAME);
+	}
+
+	@After
+	public void cleanUp() {
+		mongoTemplate.dropCollection(COLLECTION_NAME);
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	@Test
+	@MongoDbAvailable
+	public void testNoFactorySpecified() {
+		MongoDbFactory nullFactory = null;
+
+		try {
+			new MongoDbOutboundGateway(nullFactory);
+			Assert.fail("Expected the test case to throw an IllegalArgumentException");
+		}
+		catch (IllegalArgumentException e) {
+			assertEquals("MongoDbFactory translator must not be null!", e.getMessage());
+		}
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	@Test
+	@MongoDbAvailable
+	public void testNoTemplateSpecified() {
+		MongoOperations mongoTemplate = null;
+
+		try {
+			new MongoDbOutboundGateway(mongoTemplate);
+			Assert.fail("Expected the test case to throw an IllegalArgumentException");
+		}
+		catch (IllegalArgumentException e) {
+			assertEquals("mongoTemplate must not be null.", e.getMessage());
+		}
+	}
+
+	@Test
+	@MongoDbAvailable
+	public void testNoQuerySpecified() {
+		Message<String> message = MessageBuilder.withPayload("test").build();
+		MongoDbOutboundGateway gateway = createGateway();
+
+		try {
+			gateway.afterPropertiesSet();
+			gateway.handleRequestMessage(message);
+			Assert.fail("Expected the test case to throw an IllegalArgumentException");
+		}
+		catch (IllegalArgumentException e) {
+			assertEquals("no query specified", e.getMessage());
+		}
+	}
+
+	@Test
+	@MongoDbAvailable
+	public void testListOfResultsWithQueryExpressionAndLimit() {
+		Message<String> message = MessageBuilder.withPayload("").build();
+		MongoDbOutboundGateway gateway = createGateway();
+		gateway.setQueryExpression(
+				PARSER.parseExpression("new BasicQuery('{''address.state'' : ''PA''}').limit(2)"));
+		gateway.afterPropertiesSet();
+
+		Object result = gateway.handleRequestMessage(message);
+
+		List<Person> persons = getPersonsFromResult(result);
+		assertEquals(2, persons.size());
+	}
+
+	@Test
+	@MongoDbAvailable
+	public void testListOfResultsWithQueryFunction() {
+		Message<String> message = MessageBuilder.withPayload("Xavi").build();
+		MongoDbOutboundGateway gateway = createGateway();
+		Function<Message<String>, Query> queryFunction = msg -> new BasicQuery("{'name' : '" + msg.getPayload() + "'}");
+		FunctionExpression<Message<String>> functionExpression = new FunctionExpression<>(queryFunction);
+		gateway.setQueryExpression(functionExpression);
+		gateway.setExpectSingleResult(true);
+		gateway.setEntityClass(Person.class);
+		gateway.afterPropertiesSet();
+
+		Object result = gateway.handleRequestMessage(message);
+
+		Person person = (Person) result;
+
+		assertEquals("Xavi", person.getName());
+	}
+
+	@Test(expected = IllegalArgumentException.class)
```

Comment:
I would like to know the reason of this exception.
Therefore like in other tests - `try...catch` or consider to use `ExpectedException` `@Rule`